### PR TITLE
Add tags to module

### PR DESCRIPTION
**Pull request by antarus**
close [#2101](https://github.com/jhipster/jhipster-lite/issues/2101)

### CODE REVIEW

**Comment by antarus** [comment]
I don't have any error locally on reactapp.

File: src/main/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTag.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+record JHipsterModuleTag(String tag) {
+  private static final Pattern TAG_FORMAT = Pattern.compile("^[a-z1-9-]+$");
+  public JHipsterModuleTag {
+    Assert.field("tag", tag).notNull().noWhitespace().maxLength(50);
+
+    assertFormat(tag);
+  }
+
+  private static void assertFormat(String slug) {
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTag.java:14]: slug -> tag

File: src/main/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTags.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  private JHipsterModuleTags(JHipsterModuleTagsBuilder builder) {
+    tags = builder.tags;
+  }
+
+  static JHipsterModuleTagsBuilder builder(JHipsterModuleBuilder module) {
+    return new JHipsterModuleTagsBuilder(module);
+  }
+
+  public Collection<JHipsterModuleTag> get() {
+    return Collections.unmodifiableCollection(tags);
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTags.java:22]: You can make it unmodifiable in constructor

File: src/test/java/tech/jhipster/lite/error/domain/AssertTest.java [hunk 3]
```diff
@@ -56,6 +56,19 @@ void shouldNotValidateTabString() {
...
+  @Test
+  void shouldNotValidateStringWithWhitespace() {
+    assertThatThrownBy(() -> Assert.noWhitespace("field", "my tag"))
+      .isExactlyInstanceOf(MissingMandatoryValueException.class)
+      .hasMessageContaining("\"field\"")
+      .hasMessageContaining("(contains whitespace)");
+  }
+
+  @Test
+  void shouldValidateStringWithNoWhitespace() {
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/error/domain/AssertTest.java:68]: WithNo -> Without

File: src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+class JHipsterModuleTest {
+
+  @Test
+  void shouldTagExist() {
+    var module = JHipsterModulesFixture.module();
+    assertThat(module.tags()).isNotEmpty().contains(new JHipsterModuleTag("spring"), new JHipsterModuleTag("database"));
+  }
+
+  @Test
+  void shouldTagNotBeModifiable() {
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTest.java:18]: shouldTagNotBeModifiable -> shouldNotAllowTagUpdates or shouldHaveUnmodifiableTags or shouldHaveImmutableTags

File: src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTest.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    var tags = JHipsterModulesFixture.module().tags();
+    var tag = new JHipsterModuleTag("node");
+    assertThatThrownBy(() -> tags.add(tag)).isInstanceOf(UnsupportedOperationException.class);
+  }
+
+  @Test
+  void shouldTagNotNullAndEmpty() {
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTest.java:25]: This needs a rename :)

Comment (DamnClin) [reply src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTest.java:25]: shouldGetEmptyTagsWithoutTags?

File: src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTest.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    var module = JHipsterModulesFixture.emptyModuleBuilder().build();
+    assertThat(module.tags()).isNotNull();
+    assertThat(module.tags()).isEmpty();
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterModuleTest.java:28]: you can chain assertions + I think isEmpty() fail on null values (but to be verified)

File: src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterTagTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import org.junit.jupiter.params.provider.ValueSource;
+import tech.jhipster.lite.UnitTest;
+import tech.jhipster.lite.error.domain.MissingMandatoryValueException;
+import tech.jhipster.lite.error.domain.StringTooLongException;
+
+@UnitTest
+class JHipsterTagTest {
+
+  @Test
+  void shouldIsIdentical() {
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterTagTest.java:18]: To delete, this is not testing anything :)

File: src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterTagsTest.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+class JHipsterTagsTest {
+
+  @Test
+  void shouldIsIdentical() {
+    JHipsterModuleTags module = JHipsterModuleTags.builder(JHipsterModulesFixture.emptyModuleBuilder()).add("mytag").build();
+    assertThat(module.get()).isNotEmpty().containsOnly(new JHipsterModuleTag("mytag"));
+  }
+
+  @Test
+  void shouldNotBeValidWithStringNull() {
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterTagsTest.java:20]: I think you are testing the same think twice (should be tag responsability)

File: src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterTagsTest.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    JHipsterModuleTags.JHipsterModuleTagsBuilder builder = JHipsterModuleTags.builder(JHipsterModulesFixture.emptyModuleBuilder());
+    assertThatThrownBy(() -> builder.add((String) null).build()).isInstanceOf(MissingMandatoryValueException.class);
+  }
+
+  @Test
+  void shouldNotBeValidWithModuleTagNull() {
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/module/domain/JHipsterTagsTest.java:26]: I think you are testing the same think twice

File: src/main/java/tech/jhipster/lite/error/domain/Assert.java [hunk 10]
```diff
@@ -309,6 +325,24 @@ public StringAsserter notBlank() {
...
+    /**
+     * Ensure  that the value does not contain whitespace
+     *
+     * @return The current asserter
+     * @throws MissingMandatoryValueException
+     *           if the value contain whitespace
+     */
+    public StringAsserter noWhitespace() {
+      notNull();
+      Pattern pattern = Pattern.compile("\\s");
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/error/domain/Assert.java:337]: Extract pattern to constant (patterns are thread safe)

File: src/main/java/tech/jhipster/lite/error/domain/Assert.java [hunk 11]
```diff
@@ -309,6 +325,24 @@ public StringAsserter notBlank() {
...
+      Matcher matcher = pattern.matcher(value);
+      if (matcher.find()) {
+        throw MissingMandatoryValueException.forWhitespaceValue(field);
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/error/domain/Assert.java:340]: I don't think this is a missing mandatory value, I think this needs a dedicated exception

**Comment by DamnClin** [comment]
Need to add those tags in Rest response (so we can use them in the frontend)

File: src/main/java/tech/jhipster/lite/error/domain/Assert.java (hunk 11 above)
Comment (antarus) [reply src/main/java/tech/jhipster/lite/error/domain/Assert.java:340]: A UnauthorizedValueException perhaps?

Comment (DamnClin) [reply src/main/java/tech/jhipster/lite/error/domain/Assert.java:340]: For me you can be very specific on that, StringWithWitespacesException sounds good

**Comment by antarus** [comment]
I wonder if it wouldn't be better if the tags were managed at the RestJHipsterModule and JHipsterModuleResource level rather than JhipsterModule? tags do not add anything to jhipster module or else I have to find a way to access jhipsterModule but it is not yet build when the resource is created

**Comment by DamnClin** [comment]
Yep, I think it's better to keep it in JHipsterModuleResource

**Comment by pascalgrimaud** [comment]
Is it ready to be merged @antarus ?

**Comment by antarus** [comment]
No,i need to work on it

File: src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+      return this;
+    }
+
+    public JHipsterModuleTagsBuilder add(String tag) {
+      tags.add(new JHipsterModuleTag(tag));
+
+      return this;
+    }
+
+    public JHipsterModuleTagsBuilder add(String[] tags) {
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java:43]: Perhaps put an ellipsis here (String... tags)

File: src/main/java/tech/jhipster/lite/module/infrastructure/primary/JHipsterModuleResource.java [hunk 13]
```diff
@@ -11,6 +12,9 @@ public class JHipsterModuleResource {
   private final JHipsterModuleSlug slug;
   private final JHipsterModulePropertiesDefinition propertiesDefinition;
   private final JHipsterModuleApiDoc apiDoc;
+
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/module/infrastructure/primary/JHipsterModuleResource.java:15]: Remove line break

File: src/main/java/tech/jhipster/lite/module/infrastructure/primary/RestJHipsterModule.java [hunk 14]
```diff
@@ -38,12 +46,19 @@ public RestJHipsterModulePropertiesDefinition getProperties() {
...
+    return tags;
+  }
+
   private static class RestJHipsterModuleBuilder {
 
     private String slug;
     private String description;
     private RestJHipsterModulePropertiesDefinition properties;
 
+    private Collection<String> tags = new ArrayList<>();
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/module/infrastructure/primary/RestJHipsterModule.java:60]: Init not needed, your domain will ensure that it's empty if none set

File: src/test/java/tech/jhipster/lite/module/domain/JHipsterTagTest.java [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+import tech.jhipster.lite.error.domain.StringTooLongException;
+import tech.jhipster.lite.error.domain.StringWithWitespacesException;
+
+@UnitTest
+class JHipsterTagTest {
+
+  @Test
+  void shouldIsIdentical() {
+    JHipsterModuleTag tag = new JHipsterModuleTag("mytag");
+    assertThat(tag.tag()).isEqualTo("mytag");
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/module/domain/JHipsterTagTest.java:21]: For modules with a single field I like to create a `.get()` method reason explained here https://blog.ippon.fr/2021/04/02/jouons-avec-les-records-java/ (in french), and examples in `JHipsterProjectName`, `JHipsterPropertyDescription`, `JHipsterProjectFolder`, ...

File: src/main/java/tech/jhipster/lite/generator/client/vue/core/infrastructure/primary/VueModuleConfiguration.java [hunk 16]
```diff
@@ -18,6 +19,7 @@ JHipsterModuleResource vueModule(VueApplicationService vue) {
       .slug("vue")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addIndentation().build())
       .apiDoc(new JHipsterModuleApiDoc("Vue", "Add Vue+Vite"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/client/vue/core/infrastructure/primary/VueModuleConfiguration.java:22]: "client", "init", "vue"

File: src/main/java/tech/jhipster/lite/generator/client/vue/core/infrastructure/primary/VueModuleConfiguration.java [hunk 17]
```diff
@@ -29,6 +31,7 @@ JHipsterModuleResource vuePiniaModule(VueApplicationService vue) {
       .slug("vue-pinia")
       .withoutProperties()
       .apiDoc(new JHipsterModuleApiDoc("Vue", "Add pinia for state management"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/client/vue/core/infrastructure/primary/VueModuleConfiguration.java:34]: "client", "vue", "store"

File: src/main/java/tech/jhipster/lite/generator/infinitest/infrastructure/primary/InfinitestConfiguration.java [hunk 18]
```diff
@@ -17,6 +18,7 @@ JHipsterModuleResource infinitestModule(InfinitestApplicationService infinitest)
       .slug("infinitest-filters")
       .withoutProperties()
       .apiDoc(new JHipsterModuleApiDoc("Base", "Add filter for infinitest"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/infinitest/infrastructure/primary/InfinitestConfiguration.java:21]: "server", "init", "test"

File: src/main/java/tech/jhipster/lite/generator/init/infrastructure/primary/InitModuleConfiguration.java [hunk 19]
```diff
@@ -19,6 +20,7 @@ JHipsterModuleResource initModule(InitApplicationService inits) {
       .slug("init")
       .propertiesDefinition(initPropertiesDefinition())
       .apiDoc(new JHipsterModuleApiDoc("Init", "Init project"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/init/infrastructure/primary/InitModuleConfiguration.java:23]: "server", "init"

File: src/main/java/tech/jhipster/lite/generator/init/infrastructure/primary/InitModuleConfiguration.java [hunk 20]
```diff
@@ -30,6 +32,7 @@ JHipsterModuleResource initMinimalModule(InitApplicationService inits) {
       .slug("init-minimal")
       .propertiesDefinition(initPropertiesDefinition())
       .apiDoc(new JHipsterModuleApiDoc("Init", "Init minimal project"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/init/infrastructure/primary/InitModuleConfiguration.java:35]: "server", "init"

File: src/main/java/tech/jhipster/lite/generator/server/hexagonaldocumentation/infrastructure/primary/HexagonalArchitectureDocumentationModuleConfiguration.java [hunk 21]
```diff
@@ -19,6 +20,7 @@ JHipsterModuleResource hexagonalArchitectureDocumentationModule(
       .slug("application-service-hexagonal-architecture-documentation")
       .withoutProperties()
       .apiDoc(new JHipsterModuleApiDoc("Spring Boot", "Add documentation for hexagonal architecture"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/hexagonaldocumentation/infrastructure/primary/HexagonalArchitectureDocumentationModuleConfiguration.java:23]: "server", "hexagonal", "architecture", "documentation"

File: src/main/java/tech/jhipster/lite/generator/server/javatool/arch/infrastructure/primary/ArchUnitModuleConfiguration.java [hunk 22]
```diff
@@ -18,6 +19,7 @@ JHipsterModuleResource archUnitModule(JavaArchUnitApplicationService archUnit) {
       .slug("java-archunit")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addBasePackage().build())
       .apiDoc(new JHipsterModuleApiDoc("Java", "Add Hexagonal Arch classes to project"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/javatool/arch/infrastructure/primary/ArchUnitModuleConfiguration.java:22]: "server", "hexagonal", "architecture", "test"

File: src/main/java/tech/jhipster/lite/generator/server/javatool/base/infrastructure/primary/JavaBaseModuleConfiguration.java [hunk 23]
```diff
@@ -18,6 +19,7 @@ JHipsterModuleResource javaBaseModule(JavaBaseApplicationService javaBase) {
       .slug("java-base")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addBasePackage().addProjectBaseName().build())
       .apiDoc(new JHipsterModuleApiDoc("Java", "Add Base classes and Error domain to project"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/javatool/base/infrastructure/primary/JavaBaseModuleConfiguration.java:22]: "server", "init"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/async/infrastructure/primary/SpringBootAsyncConfiguration.java [hunk 24]
```diff
@@ -18,6 +19,7 @@ JHipsterModuleResource springBootAsync(SpringBootAsyncApplicationService springB
       .slug("springboot-async")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addBasePackage().addProjectBaseName().build())
       .apiDoc(new JHipsterModuleApiDoc("Spring Boot - Tools", "Add asynchronous execution and scheduling configuration"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/async/infrastructure/primary/SpringBootAsyncConfiguration.java:22]: "server", "spring", "spring-boot"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java [hunk 25]
```diff
@@ -19,6 +20,7 @@ JHipsterModuleResource jhipsterV7BannerResource(BannerApplicationService banners
       .slug("banner-jhipster-v7")
       .withoutProperties()
       .apiDoc(new JHipsterModuleApiDoc(TAG, "Add banner JHipster v7 for Angular"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java:23]: "server", "spring", "spring-boot", "init"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java [hunk 26]
```diff
@@ -30,6 +32,7 @@ JHipsterModuleResource jhipsterV7ReactBannerResource(BannerApplicationService ba
       .slug("banner-jhipster-v7-react")
       .withoutProperties()
       .apiDoc(new JHipsterModuleApiDoc(TAG, "Add banner JHipster v7 for React"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java:35]: "server", "spring", "spring-boot", "init"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java [hunk 27]
```diff
@@ -41,6 +44,7 @@ JHipsterModuleResource jhipsterV7VueBannerResource(BannerApplicationService bann
       .slug("banner-jhipster-v7-vue")
       .withoutProperties()
       .apiDoc(new JHipsterModuleApiDoc(TAG, "Add banner JHipster v7 for Vue"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java:47]: "server", "spring", "spring-boot", "init"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java [hunk 28]
```diff
@@ -52,6 +56,7 @@ JHipsterModuleResource jhipsterV2BannerResource(BannerApplicationService banners
       .slug("banner-jhipster-v2")
       .withoutProperties()
       .apiDoc(new JHipsterModuleApiDoc(TAG, "Add banner JHipster v2"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java:59]: "server", "spring", "spring-boot", "init"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java [hunk 29]
```diff
@@ -63,6 +68,7 @@ JHipsterModuleResource jhipsterV3BannerResource(BannerApplicationService banners
       .slug("banner-jhipster-v3")
       .withoutProperties()
       .apiDoc(new JHipsterModuleApiDoc(TAG, "Add banner JHipster v3"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java:71]: "server", "spring", "spring-boot", "init"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java [hunk 30]
```diff
@@ -74,6 +80,7 @@ JHipsterModuleResource jhipsterIpponBannerResource(BannerApplicationService bann
       .slug("banner-ippon")
       .withoutProperties()
       .apiDoc(new JHipsterModuleApiDoc(TAG, "Add banner for Ippon applications"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/banner/infrastructure/primary/rest/BannerResourceConfiguration.java:83]: "server", "spring", "spring-boot", "init"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/cucumber/infrastructure/primary/CucumberModuleConfiguration.java [hunk 31]
```diff
@@ -27,6 +28,7 @@ JHipsterModuleResource cucumberModule(CucumberApplicationService cucumber) {
           .build()
       )
       .apiDoc(new JHipsterModuleApiDoc("Spring Boot - Component Tests", "Add cucumber integration to project"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/cucumber/infrastructure/primary/CucumberModuleConfiguration.java:31]: "server", "spring", "spring-boot", "test"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/database/mariadb/infrastructure/primary/MariaDBModuleConfiguration.java [hunk 32]
```diff
@@ -20,6 +21,7 @@ JHipsterModuleResource mariaDBModule(MariaDBApplicationService applicationServic
       .slug("mariadb")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addBasePackage().addIndentation().addProjectBaseName().build())
       .apiDoc(new JHipsterModuleApiDoc("Spring Boot - Database", "Add MariaDB to project"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/database/mariadb/infrastructure/primary/MariaDBModuleConfiguration.java:24]: "server", "spring", "spring-boot", "database", "db"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/database/mysql/infrastructure/primary/MySQLModuleConfiguration.java [hunk 33]
```diff
@@ -20,6 +21,7 @@ JHipsterModuleResource mySQLModule(MySQLApplicationService applicationService) {
       .slug("mysql")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addBasePackage().addIndentation().addProjectBaseName().build())
       .apiDoc(new JHipsterModuleApiDoc("Spring Boot - Database", "Add MySQL to project"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/database/mysql/infrastructure/primary/MySQLModuleConfiguration.java:24]: "server", "spring", "spring-boot", "database", "db"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/database/postgresql/infrastructure/primary/PostgresqlModuleConfiguration.java [hunk 34]
```diff
@@ -20,6 +21,7 @@ JHipsterModuleResource postgresqlModule(PostgresqlApplicationService postgresql)
       .slug("postgresql")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addBasePackage().addIndentation().addProjectBaseName().build())
       .apiDoc(new JHipsterModuleApiDoc("Spring Boot - Database", "Add Postgresql to project"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/database/postgresql/infrastructure/primary/PostgresqlModuleConfiguration.java:24]: "server", "spring", "spring-boot", "database", "db"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/devtools/infrastructure/primary/DevToolsModuleConfiguration.java [hunk 35]
```diff
@@ -18,6 +19,7 @@ JHipsterModuleResource devTools(DevToolsApplicationService devtools) {
       .slug("springboot-devtools")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addBasePackage().addProjectBaseName().addIndentation().build())
       .apiDoc(new JHipsterModuleApiDoc("Spring Boot - Tools", "Add spring boot tools."))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/devtools/infrastructure/primary/DevToolsModuleConfiguration.java:22]: "server", "init"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/logsspy/infrastructure/primary/LogsSpyModuleConfiguration.java [hunk 36]
```diff
@@ -18,6 +19,7 @@ JHipsterModuleResource logsSpyModule(LogsSpyApplicationService logsSpys) {
       .slug("logs-spy")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addBasePackage().build())
       .apiDoc(new JHipsterModuleApiDoc("Spring Boot", "Add a JUnit extension to test logs"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/logsspy/infrastructure/primary/LogsSpyModuleConfiguration.java:22]: "server", "test"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/mvc/security/oauth2/infrastructure/primary/OAuth2ModuleConfiguration.java [hunk 37]
```diff
@@ -23,6 +24,7 @@ JHipsterModuleResource oAuth2Module(OAuth2SecurityApplicationService oAuth2) {
           "Add a Spring Security: OAuth 2.0 / OIDC Authentication (stateful, works with Keycloak and Okta)"
         )
       )
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/mvc/security/oauth2/infrastructure/primary/OAuth2ModuleConfiguration.java:27]: "server", "spring", "spring-boot", "authentication"

File: src/main/java/tech/jhipster/lite/generator/server/springboot/mvc/security/oauth2/infrastructure/primary/OAuth2ModuleConfiguration.java [hunk 38]
```diff
@@ -34,6 +36,7 @@ JHipsterModuleResource oAuth2AccountModule(OAuth2SecurityApplicationService oAut
       .slug("springboot-oauth2-account")
       .propertiesDefinition(JHipsterModulePropertiesDefinition.builder().addBasePackage().build())
       .apiDoc(new JHipsterModuleApiDoc("Spring Boot - MVC - Security", "Add a account context for OAuth 2.0 / OIDC Authentication"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/mvc/security/oauth2/infrastructure/primary/OAuth2ModuleConfiguration.java:39]: "server", "spring", "spring-boot", "authentication", "account", "user"

File: src/main/java/tech/jhipster/lite/generator/setup/codespaces/infrastructure/primary/CodespaceModuleConfiguration.java [hunk 39]
```diff
@@ -19,6 +20,7 @@ JHipsterModuleResource codespaceModule(CodespacesApplicationService codespaces)
       .slug("github-codespaces")
       .propertiesDefinition(propertiesDefinition())
       .apiDoc(new JHipsterModuleApiDoc("Codespaces", "Init GitHub Codespaces configuration files"))
+      .tags(JHipsterModuleTags.DEFAULT_TAGS)
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/setup/codespaces/infrastructure/primary/CodespaceModuleConfiguration.java:23]: Leave empty :)

File: src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java [hunk 40]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.List;
+import tech.jhipster.lite.error.domain.Assert;
+
+public class JHipsterModuleTags {
+
+  public static final JHipsterModuleTags DEFAULT_TAGS = builder().add("not-defined").build();
+  private final Collection<JHipsterModuleTag> tags;
+
+  private JHipsterModuleTags(JHipsterModuleTagsBuilder builder) {
+    tags = Collections.unmodifiableCollection(builder.tags);
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java:15]: I think you should sort tags alphabetically so they will be displayed in the same order everytime

**Comment by antarus** [comment]
The rework was a bit long :D

File: src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java [hunk 41]
```diff
@@ -0,0 +1,57 @@
+package tech.jhipster.lite.module.domain;
+
+import java.util.*;
+import tech.jhipster.lite.error.domain.Assert;
+
+public class JHipsterModuleTags {
+
+  public static final JHipsterModuleTags DEFAULT_TAGS = builder().add("none").build();
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java:8]: Can't this be deleted?

File: src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java [hunk 42]
```diff
@@ -0,0 +1,57 @@
...
+
+  private final Collection<JHipsterModuleTag> tags;
+
+  private JHipsterModuleTags(JHipsterModuleTagsBuilder builder) {
+    tags = Collections.unmodifiableCollection(builder.tags.stream().sorted(Comparator.comparing(JHipsterModuleTag::get)).toList());
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java:13]: You can extract the comparator in a constant

File: src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java (hunk 41 above)
Comment (antarus) [reply src/main/java/tech/jhipster/lite/module/domain/JHipsterModuleTags.java:8]: yes

**Comment by antarus** [comment]
done

**Comment by pascalgrimaud** [comment]
Should be good
Thanks @antarus for the work
Thanks @DamnClin for the review

**Pull request merged**